packages/mdc-grid-list/README.md

```diff
@@ -0,0 +1,195 @@
+        <img class="mdc-grid-tile__primary-content" src="images/1-1.jpg" />
+      </div>
+      <span class="mdc-grid-tile__secondary">
+        <span class="mdc-grid-tile__title">Title</span>
+        <span class="mdc-grid-tile__support-text">Support text</span>
+      </span>
+    </li>
+  </ul>
+</div>
+```
+
+### Add icon to secondary content (caption)
+
+You can add icon to caption by adding `mdc-grid-list--with-icon-align-start` or
+`mdc-grid-list--with-icon-align-end` and changing the markup.
+
+```html
+<div class="mdc-grid-list mdc-grid-list--with-icon-align-start">
+<!-- or aspect ratio you name it. -->
+  <ul class="mdc-grid-list__tiles">
+    <li class="mdc-grid-tile">
+      <div class="mdc-grid-tile__primary">
+        <img class="mdc-grid-tile__primary-content" src="images/1-1.jpg" />
+      </div>
+      <span class="mdc-grid-tile__secondary">
+        <i class="mdc-grid-tile__icon material-icons">star_border</i>
+        <span class="mdc-grid-tile__title">Title</span>
+      </span>
+    </li>
+  </ul>
+</div>
+```
+
+### Use div in replace of img
+
+In case you cannot ensure all your assets will have the same aspect ratio, you
+can use `div` instead of `img` markup. It will resize the assets to cover the tile
+and crop the assets to display the center part.
+
+```html
+<div class="mdc-grid-list mdc-grid-list--header-caption">
+  <ul class="mdc-grid-list__tiles">
+    <li class="mdc-grid-tile">
+      <div class="mdc-grid-tile__primary">
+        <div class="mdc-grid-tile__primary-content demo-grid-tile-content"></div>
+      </div>
+      <span class="mdc-grid-tile__secondary">
+        <span class="mdc-grid-tile__title">Title</span>
+      </span>
+    </li>
+  </ul>
+</div>
+```
+
+In this case, you should set `background-image:url(...)` instead of `src`. However,
+the method results in a less semantic markup, so we don't use this method by
+default.
+
+
+### RTL Support
+
+`mdc-grid-list` is automatically RTL-aware, and will re-position elements whenever
+it, or its ancestors, has a `dir="rtl"` attribute.
+
+
+### Theme
+
+`mdc-grid-list` support theme. Tile `mdc-grid-tile__primary` uses background and text on
+background, `mdc-grid-tile__secondary` uses the primary color and text on primary.
+
+
+### Using the Foundation Class
+
+MDCGridList ships with an `MDCGridListFoundation` class that external frameworks and libraries
+can use to build their own MDCGridList components with minimal effort. As with all foundation
+classes, an adapter object must be provided. The adapter for grid list must provide the following
+functions, with correct signatures:
+
+| Method Signature | Description |
+| --- | --- |
+| `getOffsetWidth() => number` | Get root element `mdc-grid-list` offsetWidth. |
+| `getTileOffsetWidthAtIndex(index: number) => number` | Get offsetWidth of `mdc-grid-tile` at specified index. |
+| `setTilesWidth(value: number) => void` | Set `mdc-grid-list__tiles` container width to value. |
```

Comment:
Here you actually mentioned 2 things:

1.  Instead of using `setTilesWidth`, uses `setStyleForTilesElement`. It is in fact making the adapter function more generic. I can totally be persuaded since setting a style to an element is done in the same way in most cases. I can even see potential advantages if we ever add more parameters to grid list, this can prevent user repeat themselves otherwise we may be forced to make breaking changes. Let me know you thoughts.

2. I am not sure if going with `px` string in the adapter is the right way. Consider the following situation:

**Scenario A**
Our function calls 

    this.adapter_.setStyleForTilesElement('width', tilesWidth)

*Some framework set width numerically may implement its adapter*:

    setStyleForTilesElement: (property, value) => SOME_METHOD_SET_WIDTH_WITH_NUMERICAL_VALUE(value)

*Some framework set width with px string may implement its adapter*:
    
    setStyleForTilesElement: (property, value) => SOME_METHOD_SET_WIDTH_WITH_PXSTRING(`${value}px`)

**Scenario B**
Our function calls 

    this.adapter_.setStyleForTilesElement('width',`${tilesWidth}px`) 

*Some framework set width with px string may implement its adapter*:
    
    setStyleForTilesElement: (property, pxString) => SOME_METHOD_SET_WIDTH_WITH_PXSTRING(pxString)

*Some framework set width numerically may implement its adapter*:

    setStyleForTilesElement: (property, pxString) => SOME_METHOD_SET_WIDTH_WITH_NUMERICAL_VALUE(parseInt(pxString))

Anyway, one of them need to convert between numerical value and px string. Personally I prefer the former, which is less opinionated. But if you have some argument saying that px string is more popular among frameworks, let me know.